src/cuda/1d/cufinufft1d.cu

```diff
@@ -97,20 +103,25 @@ int cufinufft1d2_exec(cuda_complex<T> *d_c, cuda_complex<T> *d_fk,
 
     d_plan->c  = d_cstart;
     d_plan->fk = d_fkstart;
-
-    // Step 1: amplify Fourier coeffs fk and copy into upsampled array fw
-    if (d_plan->opts.modeord == 0) {
-      if ((ier = cudeconvolve1d<T, 0>(d_plan, blksize))) return ier;
-    } else {
-      if ((ier = cudeconvolve1d<T, 1>(d_plan, blksize))) return ier;
+    
+    // Skip steps 1 and 2 if interponly
+    if (!d_plan->opts.gpu_spreadinterponly) {
```

Comment:
do we need this if here? There is a continue on L60, this should not be reachable.

Reply:
got it.